### DISCUSSION TITLE
Using Authorization Grant flow, I only get to a Tidal error screen

### DISCUSSION THREAD

**Discussion by gillyhl** [tidal-music/discussions#115] [Q&A] [answered]
Whenever I attempt an authorisation flow on a test app I have, I get navigated to the below screenshot.
<img width="684" alt="Screenshot 2024-11-19 at 14 49 16" src="https://github.com/user-attachments/assets/9137730f-569e-456c-9886-7a4fff558fff">

Below is how I generate the auth URL
```js
const authURL = `https://login.tidal.com/authorize?` +
        `response_type=code&` +
        `client_id=${CLIENT_ID}&` +
        `redirect_uri=${encodeURIComponent(REDIRECT_URI)}&` +
        `scope=playlists.read collection.read playlists.write` +
        `state=foobanana` +
        `code_challenge=${codeChallenge}&` +
        `code_challenge_method=S256`;
```

Strangely I get further when I don't specify a PKCE code challenge, that logs in and goes back to my redirect (albeit stating that PCKE challenge is required). Is there any further debugging I could be doing?

**Comment by srcurran** [comment, marked as answer]
I have been having the same issue & came here to post the same question.

The best I can tell is that the Auth Code Flow is currently blocked for non Tidal applications [(source)](https://github.com/tidal-music/tidal-sdk-web/tree/main/packages/auth#authorization-code-flow-user-login). I asked support for more information but was not given any details & instead was pointed here. 

I had found success with the Access Token approach, but that doesn't look like it would get me the data I am trying to pull (currently playing data for the auth'd user).

**Reply by gillyhl** [reply]
Thanks for the reply! It seems that it is the case that it must be blocked for now (not shown in the official portal docs though) So it'll be a case of sitting tight until it is enabled.

**Comment by stefanmoro** [comment]
Hey @gillyhl and @srcurran ,

It should be possible to login users using the authcode flow. I just tested, acquiring a few scopes, and it worked fine.

Please double check that you have correctly configured your client in the dashboard (scopes, redirect URI).

Also, thanks for pointing out the incorrect documentation in the SDK README, should be fixed now.

**Comment by gillyhl** [comment]
Thanks for the reply @stefanmoro.

I've got a client configured with redirect URL of `http://localhost:8000/auth/callback` (is having a non HTTPS redirect a problem?) and I have these scopes enabled in the client and these are the scopes I requested in the authorisation URL:
- `playlists.read`
- `collection.read`
- `playlists.write`

**Reply by rchl** [reply]
Can you double check that
- you are using the same client ID that has that callback URI specified
- that the client ID actually has the expected callback URI specified

**Reply by gillyhl** [reply]
I'm so sorry, I've been wasting your time, I didn't generate the auth URL well, it was missing the `&` between the scope and code challenge in the string I used (that will learn me for not just using a query stringifier from the start)

Apologies again, this is PEBCAK.
